packages/proxy/lib/http/util/cookies.ts

```diff
@@ -220,18 +220,20 @@ export class CookiesHelper {
     // cross site cookies cannot set lax/strict cookies in the browser for xhr/fetch requests (but ok with navigation/document requests)
     // NOTE: This is allowable in firefox as the default cookie behavior is no_restriction (none). However, this shouldn't
     // impact what is happening in the server-side cookie jar as Set-Cookie is still called and firefox will allow it to be set in the browser
-    if (this.request.requestedWith && this.siteContext === 'cross-site' && toughCookie.sameSite !== 'none') {
+    const isXhrOrFetchRequest = this.request.resourceType === 'fetch' || this.request.resourceType === 'xhr'
```

Comment:
resourceType can now be any valid resource, so we need to check for a fetch or xhr. This is covered by the cookie e2e tests